### PR TITLE
feat(portable-text-editor): move PTE into monorepo

### PR DESCRIPTION
**Pull request by skogsmaskin**
### Description

This will introduce a new package to the monorepo. `@sanity/portable-text-editor`. Previously the PTE lived in its' own repository (https://github.com/sanity-io/portable-text-editor), but is now part of the lerna bundle.

This will make development easier without having to do a lot of symlinking.

### What to review

All changed that are not part of the `@sanity/portable-text-editor` package.

### Notes for release

Moved @sanity/portable-text-editor into monorepo.

### CODE REVIEW

File: packages/@sanity/portable-text-editor/package.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  "types": "./lib/index.d.ts",
+  "author": "Sanity.io <hello@sanity.io>",
+  "engines": {
+    "node": ">=8.0.0"
+  },
+  "license": "MIT",
+  "scripts": {
+    "build": "ttsc --module commonjs",
+    "start": "cd demo && vite .",
+    "clean": "rimraf lib coverage",
```
Comment (bjoerge) [inline packages/@sanity/portable-text-editor/package.json:15]: should add `dist` here as well

File: packages/@sanity/portable-text-editor/package.json [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  "description": "Portable Text Editor made in React",
+  "main": "./lib/index.js",
+  "types": "./lib/index.d.ts",
+  "author": "Sanity.io <hello@sanity.io>",
+  "engines": {
+    "node": ">=8.0.0"
+  },
+  "license": "MIT",
+  "scripts": {
+    "build": "ttsc --module commonjs",
```
Comment (bjoerge) [inline packages/@sanity/portable-text-editor/package.json:13]: is this still required?

File: packages/@sanity/portable-text-editor/package.json [hunk 3]
```diff
@@ -0,0 +1,95 @@
+{
+  "name": "@sanity/portable-text-editor",
+  "version": "2.14.0",
+  "description": "Portable Text Editor made in React",
+  "main": "./lib/index.js",
+  "types": "./lib/index.d.ts",
+  "author": "Sanity.io <hello@sanity.io>",
+  "engines": {
+    "node": ">=8.0.0"
```
Comment (bjoerge) [inline packages/@sanity/portable-text-editor/package.json:9]: is this accurate? I see now that the rest of the packages in here has >= 10.0.0 which is out of LTS, so we should probably change this across packages.

File: packages/@sanity/portable-text-editor/package.json [hunk 4]
```diff
@@ -0,0 +1,95 @@
+{
+  "name": "@sanity/portable-text-editor",
+  "version": "2.14.0",
+  "description": "Portable Text Editor made in React",
+  "main": "./lib/index.js",
+  "types": "./lib/index.d.ts",
```
Comment (bjoerge) [inline packages/@sanity/portable-text-editor/package.json:6]: this should be changed to `./dist/dts/index.d.ts`

File: packages/@sanity/portable-text-editor/package.json [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  },
+  "license": "MIT",
+  "scripts": {
+    "build": "ttsc --module commonjs",
+    "start": "cd demo && vite .",
+    "clean": "rimraf lib coverage",
+    "posttest": "tsc --noEmit --emitDeclarationOnly false",
+    "test:watch": "jest --watch",
+    "test": "jest",
+    "watch": "ttsc --module commonjs --watch",
```
Comment (bjoerge) [inline packages/@sanity/portable-text-editor/package.json:19]: is this still accurate?

File: packages/@sanity/portable-text-editor/package.json [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    "gen-doc": "typedoc --out docs src",
+    "prettier": "prettier --write './**/*.{ts,tsx,js,css,html}'",
+    "eslint": "eslint 'src/**/*.{ts,js,tsx}'"
+  },
+  "files": [
+    "lib/**/*"
```
Comment (bjoerge) [inline packages/@sanity/portable-text-editor/package.json:25]: I think `dist/**/*` needs to be included here as well

File: packages/@sanity/portable-text-editor/tslint.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (bjoerge) [inline packages/@sanity/portable-text-editor/tslint.json:1]: tslint is deprecated - can we use eslint instead?

File: packages/@sanity/portable-text-editor/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,17 @@
+{
+  "extends": "../../../tsconfig",
+  "include": ["src/**/*"],
+  "compilerOptions": {
+    "composite": true,
+    "rootDir": "./src",
+    "outDir": "./dist/dts",
+    // todo: we need to disable strict in a transition phase and should be removed eventually
```
Comment (bjoerge) [inline packages/@sanity/portable-text-editor/tsconfig.json:8]: is it intentional to keep these commented out? maybe remove if possible?

**Review by bjoerge — CHANGES_REQUESTED** [review]
I left some comments that would be good to take a look at.

Note: I focused mainly on the package configuration in this review and have not been looking at the actual code imported from the external repo.

File: packages/@sanity/portable-text-editor/package.json (hunk 6 above)
Comment (skogsmaskin) [reply packages/@sanity/portable-text-editor/package.json:25]: Looks like no other packages has this property. Removing it all together.

File: packages/@sanity/form-builder/tsconfig.json [hunk 9]
```diff
@@ -3,20 +3,21 @@
...
-    "strictNullChecks": false,
+    "noImplicitThis": false,
+    "outDir": "./dist/dts",
+    "rootDir": "./src",
     "strictFunctionTypes": false,
+    "strictNullChecks": false,
     "strictPropertyInitialization": false,
-    "noImplicitThis": false,
-    "jsx": "react"
+    // todo: we need to disable strict in a transition phase and should be removed eventually
```
[the remark this answers is not in the document]
Comment (skogsmaskin) [reply packages/@sanity/form-builder/tsconfig.json]: Removed the changes

File: packages/@sanity/portable-text-editor/.babelrc [hunk 10]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../../../.babelrc",
+  "presets": ["@babel/preset-react", "@babel/preset-typescript"],
+  "env": {
+    "test": {
+      "plugins": ["transform-es2015-modules-commonjs", "@babel/plugin-proposal-class-properties"]
```
[the remark this answers is not in the document]
Comment (skogsmaskin) [reply packages/@sanity/portable-text-editor/.babelrc]: Not needed anymore.

File: packages/@sanity/portable-text-editor/package.json (hunk 4 above)
Comment (skogsmaskin) [reply packages/@sanity/portable-text-editor/package.json]: Fixed!

File: packages/@sanity/portable-text-editor/tsconfig.json (hunk 8 above)
Comment (skogsmaskin) [reply packages/@sanity/portable-text-editor/tsconfig.json]: Cleaned up.

File: packages/@sanity/portable-text-editor/tslint.json (hunk 7 above)
Comment (skogsmaskin) [reply packages/@sanity/portable-text-editor/tslint.json]: I guess we already do? Removing it.

**Comment by skogsmaskin** [comment]
> I left some comments that would be good to take a look at.
> 
> Note: I focused mainly on the package configuration in this review and have not been looking at the actual code imported from the external repo.

Thank you! I think all the feedback should be taken into account now.

**Review by bjoerge — APPROVED** [review]
Great stuff! ✨

**Comment by skogsmaskin** [comment]
Thanks! I've cleaned up the commit history now.

**Pull request merged**